common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/DiagnosisKeyService.java

```diff
@@ -61,9 +63,28 @@ public DiagnosisKeyService(DiagnosisKeyRepository keyRepository) {
   @Transactional
   public void saveDiagnosisKeys(Collection<DiagnosisKey> diagnosisKeys) {
     for (DiagnosisKey diagnosisKey : diagnosisKeys) {
-      keyRepository.saveDoNothingOnConflict(
+      keyRepository.saveDoNothingOnConflictCountries(
           diagnosisKey.getKeyData(), diagnosisKey.getRollingStartIntervalNumber(), diagnosisKey.getRollingPeriod(),
-          diagnosisKey.getSubmissionTimestamp(), diagnosisKey.getTransmissionRiskLevel());
+          diagnosisKey.getSubmissionTimestamp(), diagnosisKey.getTransmissionRiskLevel(),
+          diagnosisKey.getOriginCountry(), diagnosisKey.getVisitedCountries().toArray(new String[0]));
+    }
+  }
+
+  /**
+   * Persists the specified collection of {@link DiagnosisKey} instances. If the key data of a particular diagnosis key
+   * already exists in the database, this diagnosis key is not persisted.
+   *
+   * @param diagnosisKeys must not contain {@literal null}.
+   * @throws IllegalArgumentException in case the given collection contains {@literal null}.
+   */
+  @Timed
+  @Transactional
+  public void saveDiagnosisKeysWithCountry(Collection<DiagnosisKey> diagnosisKeys) {
```

Comment:
Is the special variant needed or could we replace the existing save method compeltely?